### PR TITLE
Add EvaluateZeroTopN rule

### PR DESCRIPTION
**Pull request by kabunchi**

### CODE REVIEW

**Comment by Praveen2112** [comment]
Can we change the commit message to "Add EvaluateZeroTopN rule" or to "Add EvaluateTopZero rule" ?

File: presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestEvaluateZeroTopN.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                                0,
+                                ImmutableList.of(p.symbol("a")),
+                                p.filter(
+                                        expression("b > 5"),
+                                        p.values(
+                                                ImmutableList.of(p.symbol("a"), p.symbol("b")),
+                                                ImmutableList.of(
+                                                        expressions("1", "10"),
+                                                        expressions("2", "11"))))))
+                // TODO: verify contents
```
Comment (kokosing) [inline presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestEvaluateZeroTopN.java]: please solve this TODO

Comment (kabunchi) [reply presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestEvaluateZeroTopN.java]: I actually copied the TestEvaluateZeroLimit test, including this TODO.
Not sure I get what it means, will be happy to fix both if you can elaborate...
thanks.

Comment (martint) [reply presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestEvaluateZeroTopN.java]: This refers to verifying the contents of the resulting Values node. The rule testing framework doesn't yet support that, so it's fine to leave it as is. We can tackle that as a separate change later.

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**